ec2.template

```diff
@@ -354,8 +383,112 @@
         }
       }
     },
-    "AutoScalingGroup": {
+    "AutoScalingGroupPublic": {
+      "Type": "AWS::AutoScaling::AutoScalingGroup",
+      "Condition": "SubnetLocationPublic",
+      "AssociatePublicIpAddress": "true",
```

Comment:
So just realized that this is the wrong place to put it in, this particular option only works in `AWS::AutoScaling::LaunchConfiguration` will do some testing and fix up this PR to reflect that